gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveDataset.java

```diff
@@ -220,6 +353,24 @@ private Path resolvePath(String pattern) {
     }
   }
 
+  private void checkTableCompatibility(Table sourceTable, Table targetTable) throws IOException {
```

Comment:
Shall we also check other metadata, such as `Input/OutputFormat`, `Parameters`, `Privilege`, etc. You can find the list of metadata in the hive copy design doc. I think we should have a customized `PartitionComparator`.